sphinxcontrib/autoprogram.py

```diff
@@ -8,8 +9,6 @@
     :license: BSD, see LICENSE for details.
 
 """
-from __future__ import annotations
```

Comment:
This is required by 3.7+ but when present Python 3.6+ is mad and we can't use a conditional:

```
SyntaxError: from __future__ imports must occur at the beginning of the file
```

The above is raised when having:

```
import sys
if sys.version_info[0:2] != (3, 6):
     # SyntaxError will be raised on Python 3.6 where this future feature is not
     # available. For Python 3.6 the first line will make sure to enable support
     # to the annotations.
    from __future__ import annotations
```

Reply:
you could do a try/accept for this, but not sure how you are going to deal with doing `# -*- coding: future_annotations -*-` conditionally

Reply:
Does not work either, the `__future__` need to be the first non commented/docstring line of code in the file.